### PR TITLE
doctest: update to Catalan printing for Sympy 1.10 issue #1093

### PR DESCRIPTION
**Pull request by chrisjohgorman**
- Added copyright
- Fix expected return string from Sympy 1.10+ so doctest will pass
- note this change will make doctest fail for sympy < 1.10+

### CODE REVIEW

File: inst/catalan.m [hunk 1]
```diff
@@ -25,7 +26,7 @@
 %% @example
 %% @group
 %% catalan ()
-%%   @result{} (sym) Catalan
+%%   @result{} (sym) G
```
Comment (cbm755) [inline inst/catalan.m:29]: you can workaround this failing use `doctest: +SKIP_UNLESS...`, see `@sym/harmonic` for example

**Review by cbm755 — COMMENTED** [review]
Looks good so far, see comments.

I think there also might another file with this Catalan printing in the docs.

File: inst/@sym/isconstant.m [hunk 2]
```diff
@@ -24,12 +25,20 @@
...
 %%       ⎡x   1      π   ⎤
 %%       ⎢               ⎥
 %%       ⎣2  2⋅y  Catalan⎦
+%% @c doctest: +SKIP_UNLESS(pycall_sympy__ ('return Version(spver) >= Version("1.10")'))
+%% syms x y
+%% A = [x 1 pi; 2 2*y catalan()]
+%%   @result{} A = (sym 2×3 matrix)
+%%      ⎡x   1   π⎤
+%%      ⎢         ⎥
+%%      ⎣2  2⋅y  G⎦
```
Comment (cbm755) [inline inst/@sym/isconstant.m:41]: yes right idea, but this `@c doctest:` is not a conditional include.

The primary point of the docs is to be human-readable (`help @sym/constant`).

I think you should just have the `>=` case.  That is, in most cases, we assume the reader is using the latest SymPy.

(This is somewhat dealt with on [the wiki](https://github.com/cbm755/octsympy/wiki/Development-Guide), but probably not clearly enough.)

**Comment by chrisjohgorman** [comment]
I think I get what you mean.  I will redo this this evening.

**Comment by chrisjohgorman** [comment]
I have updated catalan.m and @sym/isconstant.m removing the duplicate doctests.  These three changes together do the following

- update copyright
- add @c doctest: +SKIP_UNLESS call to catalan.m disabling the test for sympy less than 3.10
- update result for catalan() call
- add two @c doctest: +SKIP_UNLESS calls to @sym/isconstant.m disabling the test for sympy less than 3.10
- update result for A = [x 1 pi; 2 2*y catalan()] call

**Comment by cbm755** [comment]
I still don't have our CI working properly (#1098) but this looks good to me!

**Pull request merged**